insights/src/main/java/org/jboss/as/insights/extension/InsightsService.java

```diff
@@ -0,0 +1,139 @@
+    public static final String PROXY_PORT = "proxyPort";
+    public static final String INSIGHTS_ENDPOINT = "insightsEndpoint";
+    public static final String SYSTEM_ENDPOINT = "systemEndpoint";
+    public static final String USER_AGENT = "userAgent";
+
+    public static final String JBOSS_PROPERTY_DIR = "jboss.server.data.dir";
+
+    public static final String INSIGHTS_PROPERTY_FILE_NAME = "insights.properties";
+
+    public static final String INSIGHTS_DESCRIPTION = "Properties file consisting of RHN login information and insights/insights URL";
+
+    public static final int DEFAULT_SCHEDULE_INTERVAL = 1;
+    public static final String DEFAULT_BASE_URL = "https://api.access.redhat.com";
+    public static final String DEFAULT_INSIGHTS_ENDPOINT = "/r/insights/v1/uploads/";
+    public static final String DEFAULT_SYSTEM_ENDPOINT = "/r/insights/v1/systems/";
+    public static final String DEFAULT_USER_AGENT = "redhat-support-lib-java";
+
+    private InsightsJdrScheduler scheduler;
+    private boolean enabled = true;
+    private ScheduledExecutorService executor;
+
+    private InjectedValue<JdrReportCollector> jdrCollector = new InjectedValue<>();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void start(StartContext context) throws StartException {
+        try {
+            if (enabled) {
+                scheduler.startScheduler();
+            }
+        } catch (IllegalStateException e) {
+            throw new StartException(e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void stop(StopContext context) {
+        final InsightsJdrScheduler scheduler = this.scheduler;
+        this.scheduler = null;
+        scheduler.stopScheduler();
+        this.executor.shutdownNow();
+    }
+
+    private InsightsService(ScheduledExecutorService executor, boolean enabled, InsightsConfiguration config) {
+        this.executor = executor;
+        this.enabled = enabled;
+        this.scheduler = new InsightsJdrScheduler(executor, config, jdrCollector);
+    }
+
+    public static  ServiceController<InsightsScheduler> addService(ServiceTarget serviceTarget, ScheduledExecutorService executor, boolean enabled,
+            InsightsConfiguration config) {
+        InsightsService service = new InsightsService(executor, enabled, config);
+        return serviceTarget.addService(SERVICE_NAME, service)
+                .addDependency(JdrReportService.SERVICE_NAME, JdrReportCollector.class, service.getJdrReportCollector())
+                .setInitialMode(ServiceController.Mode.ACTIVE)
+                .install();
+    }
+
+    @Override
+    public InsightsScheduler getValue() throws IllegalStateException, IllegalArgumentException {
+        return scheduler;
+    }
+
+    void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        if (enabled) {
+            ROOT_LOGGER.insightsEnabled();
+            try {
+                start(null);
```

Comment:
You should use a separate, private method for this, and have the start(StartContext...) method call to it. The "public synchronized void start(StartContext context) throws StartException" method is part of the MSC ServiceContract and trying to resuse it for non-MSC callers is likely to lead to bugs. For example, if you move the construction of "scheduler" to the start method per my comment above, you problem wouldn't want to reconstruct that each time setEnabled is called.